tests/dda/upload/test_wrapper.py

```diff
@@ -1,26 +1,101 @@
 import os
+import tempfile
 import uuid
 
 import pytest
 
 from dagshub.upload import Repo
+from tests.dda.mock_api import MockApi
 
 
 @pytest.fixture(scope="function")
-def test_file():
+def test_file() -> str:
     filepath = f"{uuid.uuid4()}.txt"
     with open(filepath, "w") as f:
         f.write("test data")
     yield filepath
     try:
         os.remove(filepath)
-    except OSError:
+    except (OSError, FileNotFoundError):
         pass
 
 
-def test_upload_dataset_closes_files(mock_api, upload_repo: Repo, test_file: str):
+@pytest.fixture(scope="function")
+def test_dirs() -> str:
+    folder_path = os.path.join("nested", "folder")
+    os.makedirs(folder_path, exist_ok=True)
+
+    filepath1 = os.path.join(folder_path, f"{uuid.uuid4()}.txt")
+    with open(filepath1, "w") as f:
+        f.write("test data")
+    filepath2 = os.path.join(folder_path, f"{uuid.uuid4()}.txt")
+    with open(filepath2, "w") as f:
+        f.write("test data")
+
+    yield folder_path
+    try:
+        os.remove(filepath1)
+        os.remove(filepath2)
+        os.removedirs(folder_path)
+    except (OSError, FileNotFoundError):
+        pass
```

Comment:
You can use the `tmpdir` fixture and let go of all this cleanup code, pytest will manage it for you (note the paragraph in the end, it does persist a couple test runs)
https://docs.pytest.org/en/6.2.x/tmpdir.html

Reply:
That's nice but if it's alright with you I'd rather avoid it.
It's waaaay too magical. Basically impossible to figure out where it's coming from.

Reply:
for sure, just a thing to check out